core/api/src/jsonrpc/web3_types.rs

```diff
@@ -336,6 +336,51 @@ pub struct Web3CallRequest {
     pub max_priority_fee_per_gas: Option<U256>,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum BlockParameter {
+  Specifier(BlockSpecifier),
+  Id(BlockId),
+}
+
+impl<'a> Deserialize<'a> for BlockParameter {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'a>,
+    {
+        let s = String::deserialize(deserializer)?;
+        match serde_json::from_str::<BlockSpecifier>(&s) {
+            Ok(specifier) => Ok(BlockParameter::Specifier(specifier)),
+            Err(_) => match serde_json::from_str::<BlockId>(&s) {
+                Ok(block_id) => Ok(BlockParameter::Id(block_id)),
+                Err(_) => Err(serde::de::Error::custom("Invalid BlockParameter")),
+            },
+        }
+    }
+}
+
+impl Serialize for BlockParameter {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match *self {
+            BlockParameter::Id(ref id) => id.serialize(serializer),
+            BlockParameter::Specifier(ref specifier) => specifier.serialize(serializer)
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct BlockSpecifier {
+  #[serde(skip_serializing_if = "Option::is_none")]
+  pub block_number: Option<u64>,
+  #[serde(skip_serializing_if = "Option::is_none")]
+  pub block_hash: Option<H256>,
+  #[serde(skip_serializing_if = "Option::is_none")]
+  pub require_canonical: Option<bool>,
```

Comment:
Axon is based BFT consensus which never fork.Whether the `require_canonical` is true or not leads to the same process. So do not need to parse this field.